core/CMakeLists.txt

```diff
@@ -1,49 +1,53 @@
 ############################################################################
 # CMakeLists.txt file for building ROOT (global) core package
 ############################################################################
-
-add_subdirectory(clib)
-add_subdirectory(clingutils)
-add_subdirectory(cont)
-add_subdirectory(dictgen)
-add_subdirectory(foundation)
-add_subdirectory(meta)
-add_subdirectory(metacling)
+# In directories listed below was used PARENT_SCOPE variables for defining headers
+# for clib, clingutils, cont, dictgen, foundation, meta, metacling, rint, textinput
+# and unix, winnt and macosx modules, that needed to be used after, in "base" directory
+# CMakeList.txt. Using CACHE INTERNAL solved a problem of resolving of variables in
+# scopes upper after introducing add_root_subdirectory().
```

Comment:
This comment seems redundant in the code as it describes an implementation challenge of the particular PR. Maybe move it to the commit message.